beacon-chain/rpc/eth/beacon/handlers_pool.go

```diff
@@ -461,11 +461,17 @@ func (s *Server) GetAttesterSlashings(w http.ResponseWriter, r *http.Request) {
 	ctx, span := trace.StartSpan(r.Context(), "beacon.GetAttesterSlashings")
 	defer span.End()
 
+	_, slashings := s.attesterSlashings(ctx, w)
+	httputil.WriteJson(w, &structs.GetAttesterSlashingsResponse{Data: slashings})
+}
+
+func (s *Server) attesterSlashings(ctx context.Context, w http.ResponseWriter) (string, []*structs.AttesterSlashing) {
 	headState, err := s.ChainInfoFetcher.HeadStateReadOnly(ctx)
 	if err != nil {
 		httputil.HandleError(w, "Could not get head state: "+err.Error(), http.StatusInternalServerError)
-		return
+		return "", nil
 	}
+	v := version.String(headState.Version())
```

Comment:
The version should be based on the version of the slashing, not the state. This is because (1) the only possible slashing versions are Phase0 and Electra, and (2) the state here can be after a fork with the slashing being before the fork.